### ISSUE TITLE
Inconsistent treatment of distribution names with parentheses

### ISSUE THREAD

**Issue by Gattocrucco** [gplepage/gvar#39]
The following shows inconsistent behavior:

```py
import gvar

# define a distribution with a parenthesis in the name
gvar.BufferDict.add_distribution('a(', gvar.square)
b = gvar.BufferDict({'a((x)': 2})

# this thinks '(x' is the key
print(b.extension_pattern.match('a((x)').groups()) # -> 'a', '(x'

# these think 'x' is the key
print(b['x']) # -> 4
print(b.has_dictkey('x')) # -> True
print(b.has_dictkey('(x')) # -> False

# these find no key
print(list(b.all_keys())) # -> 'a((x)'
print(list(b.extension_keys())) # -> empty
```

I think the easiest way to fix this would be to forbid parentheses in `BufferDict.add_distribution`. I'm addressing this problem because I generate transformation names automatically based on user input, so I sanitize by removing parentheses.

gvar version: 11.11.12.

**Comment by gplepage** [comment]
Yes, I agree that forbidding parentheses is a good idea. Will update the code this week. Thanks.

**Comment by gplepage** [comment]
This is fixed in version 11.11.13.

**Comment by Gattocrucco** [comment]
Thanks.
